src/register.c

```diff
@@ -1198,12 +1211,14 @@ op_yank(oparg_T *oap, int deleting, int mess)
     y_current->y_size = yanklines;
     y_current->y_type = yanktype;   // set the yank register type
     y_current->y_width = 0;
-    y_current->y_array = lalloc_clear(sizeof(char_u *) * yanklines, TRUE);
+    y_current->y_array = lalloc_clear(sizeof(string_T) * yanklines, TRUE);
     if (y_current->y_array == NULL)
     {
 	y_current = curr;
 	return FAIL;
     }
+    for (i = 0; i < y_current->y_size; ++i)
+	y_current->y_array[i].length = 0;
```

Comment:
Is this needed? Shouldn't `lalloc_clear()` have already set them to zero?

Reply:
Good pickup thanks